app/routes/task_routes.py

```diff
@@ -1 +1,87 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import Blueprint, request
+from app.models.task import Task
+from app.db import db
+from datetime import date
+from .route_utilities import validate_model, create_model, get_models_with_filters
+import requests
+import os
+
+tasks_bp = Blueprint("tasks_bp", __name__, url_prefix="/tasks")
+
+@tasks_bp.post("")
+def create_task(): 
+    request_body = request.get_json()
+    return create_model(Task, request_body)
+
+@tasks_bp.get("/<task_id>")
+def get_single_task(task_id):
+    task = validate_model(Task, task_id)
+    return {"task":task.to_dict()}
+
+@tasks_bp.get("")
+def get_all_tasks(): 
+    return get_models_with_filters(Task, request.args)
+
+@tasks_bp.put("/<task_id>")
+def update_task(task_id):
+    task = validate_model(Task, task_id)
+    request_body = request.get_json()
+
+    task.title = request_body["title"]
+    task.description = request_body["description"]
+    db.session.commit()
+
+    response = {"task":task.to_dict()}
+    return response
+
+@tasks_bp.patch("/<task_id>/mark_complete")
+def update_task_complete(task_id):
+    task = validate_model(Task, task_id)
+
+    is_complete_check = True
+
+    if task.completed_at is None:
+        task.completed_at = date.today() 
+        is_complete_check = False
+    
+    db.session.add(task)
+    db.session.commit()
+
+    if not is_complete_check:
+        create_slack_msg(task)
+        is_complete_check = True 
```

Comment:
I think that the logic here could be modified a bit for better readability and intuitiveness. I see that you are trying to use a flag to send the slack message but is that necessary since we are using this route to mark a task complete and __always__ send a message to slack?

Reply:
That's fair, I wanted to try something different. I'll change the logic so task will always send a message.